src/snowflake/cli/api/project/definition_manager.py

```diff
@@ -100,6 +101,16 @@ def _user_definition_file_if_available(project_path: Path) -> Optional[Path]:
             DefinitionManager.USER_DEFINITION_FILENAME, project_path
         )
 
+    @functools.cached_property
+    def _project_properties(self) -> ProjectProperties:
+        return load_project(self._project_config_paths)
+
     @functools.cached_property
```

Comment:
Does this need to be cached, given the only value it depends on is cached?

Reply:
It does not need. I would like to keep it in case the implementation changes in the future, and to document for users that this is cached (without having to read internal functions)

Reply:
Given it's likely a small-ish dict, I'm ok with the memory duplication here in favour of a stable contract.